core/src/main/java/org/elasticsearch/action/fieldstats/TransportFieldStatsTransportAction.java

```diff
@@ -119,18 +119,14 @@ protected FieldStatsResponse newResponse(FieldStatsRequest request, AtomicRefere
                 while (iterator.hasNext()) {
                     Map.Entry<String, Map<String, FieldStats>> entry = iterator.next();
                     FieldStats indexConstraintFieldStats = entry.getValue().get(indexConstraint.getField());
-                    if (indexConstraintFieldStats == null) {
```

Comment:
if we want to filter this out, we should do it regardless of whether some have index constraints on that field, no?